servers/src/main/java/tachyon/master/lineage/LineageMaster.java

```diff
@@ -177,17 +146,15 @@ public void processJournalEntry(JournalEntry entry) throws IOException {
   public void start(boolean isLeader) throws IOException {
     super.start(isLeader);
     if (isLeader) {
-      mCheckpointExecutionService =
-          getExecutorService().submit(
-              new HeartbeatThread(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING,
-                  new CheckpointSchedulingExcecutor(this), mTachyonConf
-                      .getInt(Constants.MASTER_LINEAGE_CHECKPOINT_INTERVAL_MS)));
-      mRecomputeExecutionService =
-          getExecutorService().submit(
-              new HeartbeatThread(HeartbeatContext.MASTER_FILE_RECOMPUTATION,
-                  new RecomputeExecutor(new RecomputePlanner(mLineageStore, mFileSystemMaster),
-                      mFileSystemMaster), mTachyonConf
-                      .getInt(Constants.MASTER_LINEAGE_RECOMPUTE_INTERVAL_MS)));
+      mCheckpointExecutionService = getExecutorService()
```

Comment:
Correct. This change separates the async persist API from the lineage component. But checkpoint is still a lineage concept: it decides when and which lineage files to persist, and there could be other checkpoint algorithms than the default one implemented (e.g. the Edge algorithm from http://www.cs.berkeley.edu/~haoyuan/papers/2014_socc_tachyon.pdf). 

Client-side API is planned after the refactoring merged, such as the ASYNC_THROUGH write type. In fact, this PR paves the road to other exciting features .